deck-service/src/main/java/magicthebuilder/deckservice/controller/DeckController.java

```diff
@@ -0,0 +1,66 @@
+package magicthebuilder.deckservice.controller;
+
+
+import magicthebuilder.deckservice.dto.*;
+import magicthebuilder.deckservice.entity.Deck;
+import magicthebuilder.deckservice.service.DeckService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/decks")
+public class DeckController {
+
+    @Autowired
+    private DeckService deckService;
+
+    @GetMapping("/admin")
+    public List<SimpleDeckGetResponseDto> getAllDecks() {
+        return deckService.getAllDecks();
+    }
+
+    @GetMapping("")
+    public List<SimpleDeckGetResponseDto> getPublicDecks() {
+        return deckService.getPublicDecks();
+    }
+
+    @GetMapping("/{userId}/{deckID}")
+    public DetailedDeckGetResponseDto getUserDecks(@PathVariable("userId") Long userId, @PathVariable("deckID") UUID deckId) {
+        return deckService.getDeckByIdAndUserId(deckId, userId);
+    }
+
+    @PostMapping
+    public UUID addDeck(@RequestBody CreateDeckDto deckDto) {
+        Deck deck = deckService.createDeckFromDto(deckDto);
+        return deckService.addDeck(deck);
+    }
+
+    @PostMapping("/{userId}/{deckID}")
+    public DetailedDeckGetResponseDto updateDeck(@RequestBody DeckUpdateRequestDto deckDto) {
```

Comment:
I don't think so. This functionality is not idempotent, and i dont think changing it so we can use PUT here makes sense, since we would have to send whole deck each time we update it, instead of just the cards we want to add/remove

Reply:
But i would love to hear your opinion on this

Reply:
Maybe sending all cards in deck (of course just ids) wouldn't be so bad since it would give idempotency on this functionality. Hard for me to say. Maybe it's an issue we should take up with @psysiu ?

Reply:
Changed it in 817b418 but despite the fact that it is for sure "correct" approach im not sure if it's easier to use